src/main/scala/org/constellation/genesis/GenesisObservationWriter.scala

```diff
@@ -0,0 +1,69 @@
+package org.constellation.genesis
+
+import better.files.File
+import cats.data.EitherT
+import cats.effect.{Concurrent, ContextShift, Sync}
+import cats.implicits._
+import io.chrisdavenport.log4cats.Logger
+import org.constellation.domain.schema.Id
+import org.constellation.primitives.Schema.GenesisObservation
+import org.constellation.serializer.KryoSerializer
+import org.constellation.storage.external.CloudStorage
+import org.constellation.{ConfigUtil, DAO}
+
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success, Try}
+
+class GenesisObservationWriter[F[_]: Concurrent: Logger](
+  cloudStorage: CloudStorage[F],
+  dao: DAO,
+  executionContext: ContextShift[F],
+  fileOperationContext: ExecutionContext
+) {
+
+  def write(genesisObservation: GenesisObservation): EitherT[F, GenesisObservationWriterException, Unit] =
+    for {
+      _ <- EitherT(writeToDisk(genesisObservation))
+      _ <- EitherT.liftF(Logger[F].info("Genesis observation saved on disk"))
+
+      result <- sendToCloudIfRequired(genesisObservation)
+      _ <- EitherT.liftF(Logger[F].info(s"Genesis observation files saved on cloud : ${result.size}"))
+    } yield ()
+
+  private def writeToDisk(genesis: GenesisObservation): F[Either[GenesisObservationWriterException, File]] =
+    executionContext.evalOn(fileOperationContext) {
+      Sync[F].delay {
+        Try(getFilePathToGenesisObservation.writeByteArray(KryoSerializer.serializeAnyRef(genesis))) match {
```

Comment:
But if you want to use `Try` anyway then:

```scala
import cats.implicits._

Try(getFilePathToGenesisObservation.writeByteArray(KryoSerializer.serializeAnyRef(genesis)))
    .toEither
    .leftMap(CannotWriteGenesisObservationToDisk)
```

There is no need to manually convert it to Either and we can get rid of `case Success(value) => Right(value)` identity function

Reply:
Is it possible to pass `exception` to `leftMap` function?

Reply:
`Try(...).toEither` in general makes `Left(exception)` in case of exception.

`Either<Left, Right>`  is "right-biased" which means that:
- `map` operates only on `Right` so it is actually `rightMap`
- `leftMap` operates only on `Left` so you can use `leftMap` to "map" exceptions to another failures

So if you have such code:

```scala
myEither
    .map(doSthA)
    .map(doSthB)
    .map(doSthC)
    .leftMap(doSthWithException)
```

then if some of the `map` fails then Either converts itself to Left and `leftMap` is called. Otherwise `leftMap` won't be called if it doesn't fail

Reply:
Yeah if I understand correctly:
1. In your case with `.leftMap(CannotWriteGenesisObservationToDisk)` will show on this line  `case Left(value) => logger.error("...")` only my message in log, without exception details, because I mapped `PermissionError`, `NoEnoughtSpace` etc. => `CannotWriteGenesisObservationToDisk`, yeap?

2. In my case if I use `case Failure(exception) => Left(CannotWriteGenesisObservationToDisk(exception))`, I mapped java exception to my exception but I leave a message in my object exception `new CannotWriteGenesisObservationToDisk(s"Cannot write genesis to disk : ${exception.getMessage}")` 
will show on this line:
`logger.error("...")`  `CannotWriteGenesisObservationToDisk with exception message` 

This is the reason why I used this, because I need pass exception to my object:
```
case Failure(exception) => Left(CannotWriteGenesisObservationToDisk(exception))
``` 

so many things can go wrong during saving the file, exception message is quite important

Reply:
> In your case with .leftMap(CannotWriteGenesisObservationToDisk) will show on this line case Left(value) => logger.error("...") only my message in log, without exception details, because I mapped PermissionError, NoEnoughtSpace etc. => CannotWriteGenesisObservationToDisk, yeap?

If you call `.toEither` then you basically do:

```scala
case Failure(exception) => Left(exception)
case Success(value) => Right(value)
```

So you have `Either[Throwable, File]` which means that you have the whole Throwable on the left side, not only the error message.

Then if you do `leftMap(CannotWriteGenesisObservationToDisk)` then it makes `Either[CannotWriteGenesisObservationToDisk, File]`

Reply:
<img width="682" alt="image" src="https://user-images.githubusercontent.com/4959931/67847923-faa37200-fb03-11e9-810b-8b6ad8a9d709.png">
<img width="719" alt="image" src="https://user-images.githubusercontent.com/4959931/67847937-01ca8000-fb04-11e9-98f1-30d24dcd29df.png">